hub/api/tensor.py

```diff
@@ -5,15 +5,15 @@
 from hub.core.chunk_engine.read import read_array, read_tensor_meta
 from hub.core.chunk_engine.write import write_array
 from hub.core.typing import StorageProvider
-from hub.util.slice import merge_slices
+from hub.util.index import Index
 
 
 class Tensor:
     def __init__(
         self,
         key: str,
         provider: StorageProvider,
-        tensor_slice: slice = slice(None),
+        index: Union[int, slice, Index] = Index(),
```

Comment:
quick question: when setting a default arg as `Index()`, does this object get created for every instance? or is one object created and used across all created instance? i feel like it should be the former, the latter would be concerning. (no change requested here)

Reply:
i tried googling but i don't know how to formulate the question